src/Command/MarkMigrated.php

```diff
@@ -150,4 +176,155 @@ protected function markAllMigrated($path)
+        }
+
+        $version = $this->getTarget();
+
+        if ($this->isOnly()) {
+            if (!in_array($version, $versions)) {
+                throw new \InvalidArgumentException("Migration `$version` was not found !");
+            }
+
+            return [$version];
+        }
+
+        $lengthIncrease = $this->hasExclude() ? 0 : 1;
+        $index = array_search($version, $versions);
+
+        if ($index === false) {
+            throw new \InvalidArgumentException("Migration `$version` was not found !");
+        }
+
+        return array_slice($versions, 0, $index + $lengthIncrease);
+    }
+
+    /**
+     * Returns the target version from `--target` or from the deprecated version argument
+     *
+     * @return string Version found as target
+     */
+    protected function getTarget()
+    {
+        $target = $this->input->getOption('target');
+        return $target ? $target : $this->input->getArgument('version');
+    }
+
+    /**
+     * Checks if the $version is for all migrations
+     *
+     * @return bool Returns true if it should try to mark all versions
+     */
+    protected function isAllVersion()
+    {
+        if ($this->isUsingDeprecatedVersion()) {
+            return false;
+        }
+
+        return $this->isUsingDeprecatedAll() || $this->input->getOption('target') === null;
+    }
+
+    /**
+     * Checks if the version is using the deprecated `all`
+     *
+     * @return bool Returns true if it is using the deprecated `all` otherwise false
+     */
+    protected function isUsingDeprecatedAll()
+    {
+        $version = $this->input->getArgument('version');
+        return $version === 'all' || $version === '*';
+    }
+
+    /**
+     * Checks if the it is for only one migration
+     *
+     * @return bool Returns true if when it is trying to mark only one migration
+     */
+    protected function isOnly()
+    {
+
+        return $this->hasOnly() || $this->isUsingDeprecatedVersion();
+    }
+
+    /**
+     * Checks if the it has exclude option
+     *
+     * @return bool Returns true if `--exclude` option gets passed in otherwise false
+     */
+    protected function hasExclude()
+    {
+        return $this->input->getOption('exclude') !== null;
+    }
+
+    /**
+     * Checks if the it has only option
+     *
+     * @return bool Returns true if `--only` option gets passed in otherwise false
+     */
+    protected function hasOnly()
+    {
+        return $this->input->getOption('only') !== null;
+    }
+
+    /**
+     * Checks for the usage of deprecated VERSION as argument when not `all`
+     *
+     * @return bool True if it is using VERSION argument otherwise false
+     */
+    protected function isUsingDeprecatedVersion()
+    {
+        $version = $this->input->getArgument('version');
+        return $version && $version !== 'all' && $version !== '*';
+    }
+
+    /**
+     * It will look for an use of `--exclude` or `--only` without a `--target`
```

Comment:
We probably should add that it also checks if the `--exclude` and `--only` arguments are used at the same time.

Something like : 

```
Checks if `--exclude` or `--only` arguments are used without the `--target` argument or if the `--exclude` and `--only` arguments are used at the same time.
```